### PR TITLE
Ana/akash-testnet-qa

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Got stuck trying to fix this:

image

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f6bb35c38acbf2cee36b9c0--lunieio.netlify.app

**Comment by Bitcoinera** [comment]
So many commits :eyes: What happened here?

**Comment by faboweb** [comment]
I merged the cosmos-hub-4 branch

**Comment by Bitcoinera** [comment]
Sending transaction seems to have worked but it gets hung up (no success modal)

image

**Comment by Bitcoinera** [comment]
The receive transaction from yesterday and the send transaction from today are gone:

image

What is going on?

**Comment by Bitcoinera** [comment]
Yes, the success / failure detection for Akash transactions is definitely broken:

image

File: api/lib/reducers/cosmosV3-reducers.js [hunk 1]
```diff
@@ -107,16 +104,13 @@ function transactionReducerV2(network, transaction, reducers) {
...
-          fee.denom
-        )
+      fees = transaction.tx.auth_info.fee.amount.map((fee) => {
+        const coinLookup = network.getCoinLookup(network, fee.denom)
         return coinReducer(fee, coinLookup, network)
       })
     }
     // We do display only the transactions we support in Lunie
-    const filteredMessages = transaction.tx.body.messages.filter(
+    const filteredMessages = transaction.tx.value.msg.filter(
```
Comment (faboweb) [inline api/lib/reducers/cosmosV3-reducers.js:113]: 🙈  the truth is, this is cosmos hub 4 but the akash has still the old format here

**Comment by faboweb** [comment]
Also provides compatibility to cosmos-hub-4 which we will QA extra

File: app/src/signing/networkMessages/cosmos-transactions.js [hunk 2]
```diff
@@ -109,14 +106,12 @@ export async function getSignableObject(
 
 export async function getBroadcastableObject(
   chainMessages,
-  { accountSequence, accountNumber, gasEstimate, fee, memo },
```
Comment (Bitcoinera) [inline app/src/signing/networkMessages/cosmos-transactions.js:111]: :eyes:

File: api/lib/reducers/cosmosV3-reducers.js (hunk 1 above)
Comment (Bitcoinera) [reply api/lib/reducers/cosmosV3-reducers.js]: But I checked and Akash Testnet has the new, the `value.msg`. What is happening?

Comment (faboweb) [reply api/lib/reducers/cosmosV3-reducers.js]: no
that is the old format

File: app/src/signing/networkMessages/cosmos-transactions.js (hunk 2 above)
Comment (faboweb) [reply app/src/signing/networkMessages/cosmos-transactions.js:111]: those don't need to be in and break cosmos-hub-4 txs. I tested on gaia and it is fine

Comment (Bitcoinera) [reply app/src/signing/networkMessages/cosmos-transactions.js:111]: :ok_hand:

File: api/lib/reducers/cosmosV3-reducers.js (hunk 1 above)
Comment (Bitcoinera) [reply api/lib/reducers/cosmosV3-reducers.js]: Did you try Akash-testnet with this in?

Comment (Bitcoinera) [reply api/lib/reducers/cosmosV3-reducers.js]: OK, of course, silly question from me. It is working great of course. Amazing!

**Pull request merged**